### PR TITLE
pagination

### PR DESCRIPTION
**Pull request by younessoudouh**
-added pagination
-modified some css
Screenshot (8)

### CODE REVIEW

File: src/pages/trending/Trending.jsx [hunk 1]
```diff
@@ -1,27 +1,70 @@
...
+    data: trendingMovies,
     isLoading,
     isError,
-  } = useFetchData("movies", () =>
-    fetchTrendingMovies(
-      `https://api.themoviedb.org/3/trending/all/day?api_key=1a0525b93e71ae4a85e33829e5132266&page=1`
-    )
-  );
+  } = useQuery(["trending-movies", page], fetchTrendingMovies, {
+    keepPreviousData: true,
```
Comment (faouzioudouh) [inline src/pages/trending/Trending.jsx:17]: Why `keepPreviousData: true,`?

File: src/pages/trending/Trending.jsx [hunk 2]
```diff
@@ -1,27 +1,70 @@
...
+  });
+
+  useEffect(() => {
+    setCountPage(trendingMovies?.total_pages);
```
Comment (faouzioudouh) [inline src/pages/trending/Trending.jsx:21]: Why are you setting this in the state?

File: src/pages/trending/Trending.jsx [hunk 3]
```diff
@@ -1,27 +1,70 @@
...
+            mediaType={movie.media_type}
+            releaseDate={movie.release_date}
+            posterPath={movie.poster_path}
+            voteAverage={movie.vote_average}
+            name={movie.name}
+            title={movie.title}
+          />
         ))}
       </div>
+      <ReactPaginate
```
Comment (faouzioudouh) [inline src/pages/trending/Trending.jsx:53]: Why did you use a library for that, can we create our own component?

File: src/util/fetch-trending-movies.js [hunk 4]
```diff
@@ -1,5 +1,8 @@
-export const fetchTrendingMovies = async(url) => {
-    const response = await fetch(url);
+export const fetchTrendingMovies = async({ queryKey }) => {
```
Comment (faouzioudouh) [inline src/util/fetch-trending-movies.js:1]: space after `async`

File: src/pages/trending/Trending.jsx (hunk 1 above)
Comment (younessoudouh) [reply src/pages/trending/Trending.jsx:17]: when you try to fetch new data The data from the last successful fetch available so will not show loading in UI,
When the new data arrives, the previous data is swapped to show the new data.

File: src/pages/trending/Trending.jsx (hunk 3 above)
Comment (younessoudouh) [reply src/pages/trending/Trending.jsx:53]: yes we can

Comment (faouzioudouh) [reply src/pages/trending/Trending.jsx:53]: Let's do it

File: src/Componnets/movie-card/MovieCard.css [hunk 5]
```diff
@@ -47,6 +46,14 @@
     border-radius: 7px;
 }
 
+.low-rating {
+    background-color: #f50057;
+}
+
+.high-rating {
+    background-color: rgb(0, 0, 138);
```
Comment (faouzioudouh) [inline src/Componnets/movie-card/MovieCard.css:54]: use HEX

File: src/Componnets/movie-card/MovieCard.jsx [hunk 6]
```diff
@@ -6,18 +6,22 @@ export const MovieCard = ({
...
-  media_type,
-  release_date,
+  voteAverage = 5,
+  mediaType,
+  releaseDate,
 }) => {
   return (
     <div className="movie-card">
-      <span className="rate">{vote_average}</span>
+      <span
```
Comment (faouzioudouh) [inline src/Componnets/movie-card/MovieCard.jsx:15]: If no rating is available, display no rating
.

File: src/Componnets/pagination/pagination.css [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    margin-right: 10px;
+    cursor: pointer;
+}
+
+.page-active {
+    background-color: #3f51b5;
+}
+
+.page:hover {
+    background-color: rgba(0, 0, 0, .2);
```
Comment (faouzioudouh) [inline src/Componnets/pagination/pagination.css:18]: use HEX

File: src/Componnets/pagination/pagination.css [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+.page-active:hover {
+    background-color: rgba(63, 81, 181, .6);
```
Comment (faouzioudouh) [inline src/Componnets/pagination/pagination.css:22]: use HEX

File: src/pages/trending/Trending.jsx [hunk 9]
```diff
@@ -1,44 +1,58 @@
 import React from "react";
+import { useState } from "react";
+import { useQuery } from "react-query";
 import { MovieCard } from "../../Componnets/movie-card/MovieCard";
```
Comment (faouzioudouh) [inline src/pages/trending/Trending.jsx:4]: Empty line between local files and external dependencies

File: src/util/fetch-trending-movies.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+    const [, page] = queryKey;
+    const response = await fetch(
+        `https://api.themoviedb.org/3/trending/all/day?api_key=1a0525b93e71ae4a85e33829e5132266&page=${page}}`
+    );
+    const data = await response.json();
+    return data;
+};
```
Comment (faouzioudouh) [inline src/util/fetch-trending-movies.js:8]: Can you convert this to custom hook?

File: src/App.jsx [hunk 11]
```diff
@@ -1,7 +1,8 @@
+import { Routes, Route } from "react-router-dom";
```
Comment (faouzioudouh) [inline src/App.jsx:1]: Thank you

File: src/api/api-trending.js [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import axios from "axios";
+import { useQuery } from "react-query";
+
+export const useTrendingApi = (queryKey) => {
+    const [, page] = queryKey;
+    return useQuery(
```
Comment (faouzioudouh) [inline src/api/api-trending.js:6]: Cool!

File: src/pages/trending/Trending.jsx [hunk 13]
```diff
@@ -1,22 +1,18 @@
...
-  } = useQuery(["trending-movies", currentPage], fetchTrendingMovies, {
-    keepPreviousData: true,
-  });
+  const { data, isLoading, isError } = useTrendingApi([
+    "trending-movies",
+    currentPage,
+  ]);
 
-  const countPage = trendingMovies?.total_pages;
+  const countPage = data?.data?.total_pages;
```
Comment (faouzioudouh) [inline src/pages/trending/Trending.jsx:15]: Why is this double data?

Comment (younessoudouh) [reply src/pages/trending/Trending.jsx:15]: Screenshot (9)

File: package.json [hunk 14]
```diff
@@ -11,6 +11,7 @@
         "react": "^18.1.0",
         "react-dom": "^18.1.0",
         "react-icons": "^4.3.1",
+        "react-paginate": "^8.1.3",
```
Comment (faouzioudouh) [inline package.json:14]: Can you remove this unused package?

File: src/api/api-trending.js [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+        queryKey,
```
Comment (faouzioudouh) [inline src/api/api-trending.js:7]:
```suggestion
        ['trending-movies', page],
```

File: src/api/api-trending.js [hunk 16]
```diff
@@ -0,0 +1,16 @@
+import axios from "axios";
+import { useQuery } from "react-query";
+
+export const useTrendingApi = (queryKey) => {
```
Comment (faouzioudouh) [inline src/api/api-trending.js:4]:
```suggestion
export const useTrendingApi = (page) => {
```

**Review by faouzioudouh — APPROVED** [review]

**Pull request merged**